src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,100 @@
+import { useState, useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
 
-export const PeoplePage = () => {
+import { Person } from '../types';
+import { Search } from '../types/Search';
+import { getPeople } from '../api';
+
+import { filteredList } from '../utils/filteredList';
+import { sortedList } from '../utils/sortedList';
+
+export const PeoplePage: React.FC = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isErrorResponse, setIsErrorResponse] = useState(false);
+  const [orderedPeople, setOrderedPeople] = useState<Person[]>(people);
+  const [searchParams] = useSearchParams();
+
+  const sexParams = searchParams.get(Search.sex);
+  const centuriesParams = searchParams.getAll(Search.centuries);
+  const queryParams = searchParams.get(Search.query);
+  const sortParams = searchParams.get(Search.sort);
+  const orderParams = searchParams.get(Search.order);
+
+  const peopleModify = () => {
+    setOrderedPeople(
+      filteredList(
+        sortedList(people, orderParams, sortParams),
+        sexParams,
+        queryParams,
+        centuriesParams,
+      ),
+    );
+  };
```

Comment:
you can remove additional state `orderedPeople` and add the value by using `useMemo` hook
```suggestion
  const orderedPeople = useMemo(() => {
    return filteredList(
      sortedList(people, orderParams, sortParams),
      sexParams,
      queryParams,
      centuriesParams,
    );
  }, [people, orderParams, sortParams, sexParams, queryParams, centuriesParams]);
```